pilot/pkg/model/service.go

```diff
@@ -82,12 +84,16 @@ type Service struct {
 	// by the caller)
 	Resolution Resolution
 
-	// CreationTime records the time this service was created, if available.
-	CreationTime time.Time `json:"creationTime,omitempty"`
+	// Protect concurrent ClusterVIPs read/write
+	Mutex sync.RWMutex
 
-	// Attributes contains additional attributes associated with the service
-	// used mostly by mixer and RBAC for policy enforcement purposes.
-	Attributes ServiceAttributes
+	// MeshExternal (if true) indicates that the service is external to the mesh.
+	// These services are defined using Istio's ServiceEntry spec.
+	MeshExternal bool
+
+	// MTLSReady service is injected with istio sidecar and ready to configure Istio mTLS
+	// true only if all instances of service are mTLS ready, otherwise false
+	MTLSReady bool
```

Comment:
Is this set by checking all service instances has mTLS ready label?

Reply:
@diemtvu yes, in pilot/pkg/serviceregistry/kube/controller/controller.go

Reply:
But then it shouldn't be an attribute of service, right? ServiceInstance or endpoint, may be?

Reply:
per discussion on on today's Networking Workgroup call, removing service level mtls functionality since endpoint level mtls should be available in time for 1.4